public/app/features/alerting/unified/hooks/useGroupedAlerts.ts

```diff
@@ -1,11 +1,30 @@
 import { useMemo } from 'react';
 import { AlertmanagerGroup } from 'app/plugins/datasource/alertmanager/types';
 import { Labels } from '@grafana/data';
+import { uniqBy } from 'lodash';
 
-export const useGroupedAlerts = (groups: AlertmanagerGroup[], groupBy: string[]) => {
+export const useGroupedAlerts = (groups: AlertmanagerGroup[], groupBy: string[]): AlertmanagerGroup[] => {
```

Comment:
Can we write a unit test for the `useGroupedAlerts` function? It's getting a bit complex and a unit test could help us refactor it later

Reply:
Agreed